object_storage_object.go

```diff
@@ -0,0 +1,83 @@
+package linodego
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+type ObjectStorageObjectURLCreateOptions struct {
+	Name               string `json:"name"`
+	Method             string `json:"method"`
+	ContentType        string `json:"content_type,omit_empty"`
+	ContentDisposition string `json:"content_disposition,omit_empty"`
+	ExpiresIn          *int   `json:"expires_in,omit_empty"`
+}
+
+type ObjectStorageObjectURL struct {
+	URL    string `json:"url"`
+	Exists bool   `json:"exists"`
+}
+
+type ObjectStorageObjectACLConfig struct {
+	ACL    string `json:"acl"`
+	ACLXML string `json:"acl_xml"`
```

Comment:
Should probably have a test exert the xml version

Reply:
So my original interpretation of the "custom" ACL was wrong. You can specify "custom", but you cannot include XML in a PUT request. The `acl_xml` field is only present in the responses from the API.